*: explain select for update statement (#1093)
pingcap/docs: v1.0/sql/dml.md

|`HAVING where_condition` | The `HAVING` clause and the `WHERE` clause are both used to filter the results. The `HAVING` clause filters the results of `GROUP BY`, while the `WHERE` clause filter the results before aggregation. |
|`ORDER BY` | The `ORDER BY` clause is used to sort the data in ascending or descending order, based on columns, expressions or items in the `select_expr` list.|
|`LIMIT` | The `LIMIT` clause can be used to constrain the number of rows. `LIMIT` takes one or two numeric arguments. With one argument, the argument specifies the maximum number of rows to return, the first row to return is the first row of the table by default; with two arguments, the first argument specifies the offset of the first row to return, and the second specifies the maximum number of rows to return.|

> |`FOR UPDATE` | All the data in the result sets are read-locked (Those data that matches the query conditions but are not in the result sets are not read-locked, such as the data written by other transactions after the current transaction is started.), in order to detect the concurrent updates. TiDB uses the [Optimistic Transaction Model](mysql-compatibility.md#transaction). The transaction conflicts are detected in the commit phase instead of statement execution phase. while executing the `SELECT FOR UPDATE` statement, if there are other transactions trying to update relevant data, the `SELECT FOR UPDATE` transaction will fail.|

Reviewer: ```suggestion
|`FOR UPDATE` | All the data in the result sets are read-locked (Those data that match the query conditions but do not exist in the result sets are not read-locked, such as data written by other transactions after the current transaction is started.), so that concurrent updates can be detected. TiDB uses the [Optimistic Transaction Model](mysql-compatibility.md#transaction). The transaction conflicts are detected in the commit phase instead of the statement execution phase. While `SELECT FOR UPDATE` statement is being executed, if there are other transactions trying to update relevant data, the `SELECT FOR UPDATE` transaction will fail.|
```
